src/qutip_tensorflow/core/data/inner.py

```diff
@@ -0,0 +1,77 @@
+import qutip
+from .tftensor import TfTensor
+import warnings
+
+with warnings.catch_warnings():
+    warnings.filterwarnings("ignore", category=DeprecationWarning)
+    import tensorflow as tf
+
+
+__all__ = ["inner_tftensor", "inner_op_tftensor"]
+
+
+def _check_shape_inner(left, right):
+    if (
+        (left.shape[0] != 1 and left.shape[1] != 1)  # Check left shape has 1
+        or right.shape[1] != 1  # Check right shape has 1
+        or (  # Check left and right shapes are compatible
+            (left.shape[0] != right.shape[0] and left.shape[1] != right.shape[1])
+            and (left.shape[0] != right.shape[1] and left.shape[1] != right.shape[0])
+        )
```

Comment:
This condition will never be true since it use only `and` and `left.shape` is already known to contain a `1` and `right.shape[1] = 1`.
What you want is `(right.shape[0] != left.shape[0] and right.shape[0] != left.shape[1])`.

Reply:
You are right. Tests were working perfect though so I will take a look at the tests and see if there is anything missing there.